lib/excon/connection.rb

```diff
@@ -32,6 +32,16 @@ def proxy=(new_proxy)
       @data[:proxy] = new_proxy
     end
 
+    def log
+      if @data[:instrumentor] && @data[:instrumentor].respond_to?(:logger)
+        @data[:instrumentor].logger
+      end
+    end
+    def log=(logger)
+      Excon::LoggingInstrumentor.logger = logger
+      @data[:instrumentor] = Excon::LoggingInstrumentor
```

Comment:
I'm a bit worried that this overwriting instrumentor might be unexpected, though leaving it out makes it much harder to use. Maybe this is a good reason for us to revisit the idea of being able to have an array of instrumentors instead of just one?

Reply:
An array of instrumentors seems reasonable to me, though I don't know what the potential pitfalls are of doing such a thing. Could data get mangled before it gets to my instrumentor, for instance? Just pondering.

Reply:
Good question, I wouldn't expect instrumentors to mutate data, but maybe as part of this we should at least more explicitly document that. What do you think?

Reply:
Maybe automatically clone any params as part of the interface? If possible, that is. Otherwise, yes, just document that your instrumentors should not mutate data.

Reply:
I considered that, but was concerned about the potential memory bloat...